src/ui/state_manager.hh

```diff
@@ -0,0 +1,25 @@
+#ifndef STATE_MANAGER_HH
+#define STATE_MANAGER_HH
+
+#include <config.hh>
+#include <QMutex>
+
+class StateManager: public QObject
+{
+  Q_OBJECT
+
+public:
+  StateManager( QWidget * parent, Config::Class & cfg );
+
+  void setState( QByteArray state, QByteArray geometry );
+
+private:
+  bool _dirty;
+  Config::Class & _cfg;
+  QByteArray _state;
+  QByteArray _geometry;
```

Comment:
Their actual meanings are old_state & old_geometry.

Reply:
private variables ,invisible to outside, both should be ok.